### PR TITLE
added confidence metric

### PR DESCRIPTION
**Pull request by SampsonML**
Hallucination score plotted during the scarlet2.plot.scene(show_hallucinations=True) call will now show -hallucination * src.morphology() where hallucination is defined as in Sampson/Melchior ICML

### CODE REVIEW

File: scarlet2/plot.py [hunk 1]
```diff
@@ -632,18 +647,20 @@ def sources(
...
-                vmin=-true_max, 
-                vmax=true_max,
-                cmap="RdBu_r",
+                norm=colors.SymLogNorm(linthresh=1, linscale=1,
+                        vmin=-true_max, vmax=true_max, base=10),
+                cmap="RdBu",
                 extent=extent,
                 origin="lower",
             )
             colorbar = plt.colorbar(im, ax=ax[k-skipped][panel])
```
Comment (pmelchior) [inline scarlet2/plot.py:660]: Please remove the colorbar: it's not particularly useful and creates different image dimensions for the model and the hallucination images.

File: scarlet2/plot.py [hunk 2]
```diff
@@ -632,18 +647,20 @@ def sources(
...
-            ax[k-skipped][panel].set_title("Hallucination Source {}".format(k), fontsize=f_size)
+            colorbar.set_label('hallucination score', rotation=270, labelpad=20, fontsize=f_size)
+            title = 'confidence: ' + str(jnp.round(metric,3))
```
Comment (pmelchior) [inline scarlet2/plot.py:662]: In keeping with the other labels: uppercase "Confidence", please

File: scarlet2/plot.py (hunk 1 above)
Comment (SampsonML) [reply scarlet2/plot.py]: Gone

File: scarlet2/plot.py (hunk 2 above)
Comment (SampsonML) [reply scarlet2/plot.py]: Updated

**Review by pmelchior — APPROVED** [review]
